### ISSUE TITLE
Differences between "Unsupported string offset operations" and "Illegal string offset" inspections.

### ISSUE THREAD

**Issue by iakoder** [kalessil/phpinspectionsea#893]
Just curious why $x[] is caught for both string and array|string variables but $x[$y] is only caught for string but not array|string.  (I'd much prefer that all four cases be caught so I can @noinspect away the ones I don't care about.)

```
/**
 * @param string  $x
 * @param string   $y
 */
function foo1(&$x, $y) {
    // Caught
    $x[] = 'Bar';
    // Caught
    $x[$y] = 'T';
}

/**
 * @param array|string  $x
 * @param string        $y
 */
function foo2(&$x, $y) {
    // Caught
    $x[] = 'Bar';
    // Not caught
    $x[$y] = 'T';
    echo 'Done';
}
```

Oh, never mind.  The Illegal string offset is a PhpStorm built-in inspection.  Ok, this is a feature request then for you to make your own Illegal string offset inspection that works for array|string types too.  :)

**Comment by orklah** [comment]
This one could be really useful. When I try to copy the code in example, only 1 out of the 4 case is caught

**Comment by kalessil** [comment]
Re-opening
